uaclient/security_status.py

```diff
@@ -118,15 +120,35 @@ def get_origin_for_installed_package(
     return "third-party"
 
 
-def get_update_status(service_name: str, ua_info: Dict[str, Any]) -> str:
+@lru_cache(maxsize=None)
+def _is_candidate_version(pkg: str, version: str) -> bool:
+    """Returns True if the package version is a candidate version."""
+    candidate_version = get_pkg_candidate_version(pkg, check_esm_cache=False)
+    if candidate_version:
+        return version == candidate_version
+    return False
```

Comment:
Maybe we can replace this for a one liner:
`return version == candidate_version`

Or is there a situation where `version` is None too ?

Reply:
The `get_pkg_candidate_version` function has the return type of `Optional[str]` so I believe there is a case where the candidate version returned can be `None`

Reply:
Candidate will be None when you request a package that is not in the Cache, or a package which is in the Cache but has no installable version (or is pinned to never, or things like that) so yeah for the sake of corner cases, keep the check